tests/unit/lib/deploy/test_deployer.py

```diff
@@ -213,6 +213,58 @@ def test_create_changeset_ClientErrorException_generic(self):
                 tags={"unit": "true"},
             )
 
+    def test_create_changeset_pass_through_optional_arguments_only_if_having_values(self):
```

Comment:
can you split this test case ... it validates 2 different things

Reply:
which 2 different things? it validates that `capabilities`, `role_arn` and `notification_arns` will be passed to cloudformation only if they have values, otherwise, they won't

Reply:
I mean that you can have 2 test cases, one if these properties have values, and the second one for the None values.
any way, it is a very minor comment